include/input.c

```diff
@@ -0,0 +1,82 @@
+/*
+This file is part of TicTacToe.
+
+TicTacToe is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+TicTacToe is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with TicTacToe.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "input.h"
+#if !defined(NXDK)
+#include <SDL2/SDL.h>
+#else
+#include <SDL.h>
+#endif
+
+int openedControllers = 0;
+
+void initController(SDL_GameController *controller, int *numOpenedControllers) {
+    // Indices start at 0, so if we have opened 1 controller this will start in the second index
+    for (int i = *numOpenedControllers; i < SDL_NumJoysticks(); ++i) {
+        if (SDL_IsGameController(i)) {
+            controller = SDL_GameControllerOpen(i);
+            if (controller) {
+                *numOpenedControllers++;
+                break;
+            }
+            else {
+                printf("Could not open Game Controller! Reason: %s\n", SDL_GetError());
+            }
+        }
+    }
+}
+
+void initButtonMap(GameButton buttonMap[]) {
```

Comment:
the array could be statically initialised in the header? Would also encourage to read it more as a "config" ;)